### PR TITLE
Support latents_mean and latents_std

### PR DESCRIPTION
**Pull request by haofanwang**
As title. 

This PR supports latents_mean and latents_std in the decoding stage of VAE as done in [pipeline_stable_diffusion_xl.py](https://github.com/huggingface/diffusers/blob/7db935a1414e35520b6bb87222e28b2f52d632fb/src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py#L1316) to make the pipeline compatible to models like [playgroundai/playground-v2.5-1024px-aesthetic](https://huggingface.co/playgroundai/playground-v2.5-1024px-aesthetic) that set latents_mean and latents_std in [config](https://huggingface.co/playgroundai/playground-v2.5-1024px-aesthetic/blob/main/vae/config.json#L30). For now, all other pipelines lead to grey image except for [pipeline_stable_diffusion_xl.py](https://github.com/huggingface/diffusers/blob/7db935a1414e35520b6bb87222e28b2f52d632fb/src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py#L1316).

Tested locally.

### CODE REVIEW

File: src/diffusers/pipelines/controlnet/pipeline_controlnet_sd_xl_img2img.py [hunk 1]
```diff
@@ -1611,6 +1611,21 @@ def __call__(
                 self.upcast_vae()
                 latents = latents.to(next(iter(self.vae.post_quant_conv.parameters())).dtype)
 
+             # unscale/denormalize the latents
```
Comment (patil-suraj) [inline src/diffusers/pipelines/controlnet/pipeline_controlnet_sd_xl_img2img.py:1614]:
```suggestion
            # unscale/denormalize the latents
```

File: src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl_inpaint.py [hunk 2]
```diff
@@ -1860,13 +1860,28 @@ def denoising_value_valid(dnv):
...
+                latents = latents / self.vae.config.scaling_factor
+
+            image = self.vae.decode(latents, return_dict=False)[0]
 
             # cast back to fp16 if needed
             if needs_upcasting:
                 self.vae.to(dtype=torch.float16)
         else:
-            return StableDiffusionXLPipelineOutput(images=latents)
+            image = latents
```
Comment (patil-suraj) [inline src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl_inpaint.py:1884]: why is this changed?

File: src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl_instruct_pix2pix.py [hunk 3]
```diff
@@ -1014,14 +1014,28 @@ def __call__(
...
+                latents = latents / self.vae.config.scaling_factor
+
+            image = self.vae.decode(latents, return_dict=False)[0]
 
             # cast back to fp16 if needed
             if needs_upcasting:
                 self.vae.to(dtype=torch.float16)
         else:
             image = latents
-            return StableDiffusionXLPipelineOutput(images=image)
```
Comment (patil-suraj) [inline src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl_instruct_pix2pix.py:1024]: same comment as above ?

**Review by patil-suraj — COMMENTED** [review]
Thanks a lot for the PR! Looks good, but we should leave other things un-touched in this PR, if there is some issue we a separate PR would be better.

**Comment by haofanwang** [comment]
Fixed.

File: src/diffusers/pipelines/controlnet/pipeline_controlnet_sd_xl_img2img.py [hunk 4]
```diff
@@ -1587,6 +1587,21 @@ def __call__(
...
+                    torch.tensor(self.vae.config.latents_mean).view(1, 4, 1, 1).to(latents.device, latents.dtype)
+                )
+                latents_std = (
+                    torch.tensor(self.vae.config.latents_std).view(1, 4, 1, 1).to(latents.device, latents.dtype)
+                )
+                latents = latents * latents_std / self.vae.config.scaling_factor + latents_mean
+            else:
+                latents = latents / self.vae.config.scaling_factor
+
             image = self.vae.decode(latents / self.vae.config.scaling_factor, return_dict=False)[0]
```
Comment (yiyixuxu) [inline src/diffusers/pipelines/controlnet/pipeline_controlnet_sd_xl_img2img.py:1605]:
```suggestion
            image = self.vae.decode(latents , return_dict=False)[0]
```

**Comment by yiyixuxu** [comment]
@haofanwang hey I fixed a mistake here https://github.com/huggingface/diffusers/pull/7132#discussion_r1518482166 but now the quality test fail again 
can you fix them thanks! 😛

**Comment by haofanwang** [comment]
Sure.

**Comment by haofanwang** [comment]
@yiyixuxu Formatted, pass my local check.
